test/test_gpu3.cpp

```diff
@@ -8716,6 +8716,91 @@ TEST_F(NVFuserTest, Reduction3DConstantIterationDomain) {
       executor_cache.fusion(), cg_outputs, inputs, {ref}, __LINE__, __FILE__);
 }
 
+// don't cache if the input tv is used by slice.
+// https://github.com/NVIDIA/Fuser/issues/1697
+TEST_F(NVFuserTest, AvoidCachingSliceInput) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  // values to trigger the original bug.
+  const int64_t eight = 8;
+  const int64_t twenty = 20;
+  const int64_t fiveTwelve = 512;
+  const int64_t batch_size = 128;
+  const int64_t hidden_size = 4096;
+  DataType input_dtype = DataType::Half;
+  auto tv0 = makeContigTensor(2, input_dtype);
+  auto tv1 = makeContigTensor(1, input_dtype);
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+
+  // inner persistent
+  auto tv2 = castOp(DataType::Float, tv0);
+  auto tv3 = exp(tv2);
+  auto tv4 = sum(tv3, {-1});
+  auto tv5 = broadcast(tv4, {false, true});
+  auto tv6 = div(tv3, tv5);
+
+  // reshape t1 to [8, 512*20]
+  auto tv7 = reshape(
+      tv1,
+      {
+          castOp(DataType::Index, IrBuilder::create<Val>(eight, DataType::Int)),
+          castOp(
+              DataType::Index,
+              IrBuilder::create<Val>(fiveTwelve * twenty, DataType::Int)),
+      });
+
+  // slice-1 reshape to hidden size
+  auto tv8 = slice(tv7, {0, 0}, {eight, fiveTwelve});
+  auto tv9 = reshape(
+      tv8,
+      {castOp(
+          DataType::Index,
+          IrBuilder::create<Val>(hidden_size, DataType::Int))});
+  auto tv10 = broadcast(tv9, {true, false});
+  auto tv11 = castOp(DataType::Float, tv10);
+  fusion->addOutput(tv11);
+
+  // slice-2  reshape to hidden size and link with inner persistent
+  auto tv12 = slice(tv7, {0, fiveTwelve * 3}, {eight, fiveTwelve * 4});
+  auto tv13 = reshape(
+      tv12,
+      {castOp(
+          DataType::Index,
+          IrBuilder::create<Val>(hidden_size, DataType::Int))});
```

Comment:
```suggestion
      {IrBuilder::create<Val>(hidden_size, DataType::Index)});
```

Reply:
They are not important. Let me remove all of them.